src/components/parcels/DictionaryTable.svelte

```diff
@@ -0,0 +1,205 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import type { CellEditingStoppedEvent, ICellRendererParams, ValueGetterParams } from 'ag-grid-community';
+  import { createEventDispatcher } from 'svelte';
+  import type { User } from '../../types/app';
+  import type { DataGridColumnDef, DataGridRowSelection, RowId } from '../../types/data-grid';
+  import type { DictionaryType } from '../../types/sequencing';
+  import { featurePermissions } from '../../utilities/permissions';
+  import DataGridActions from '../ui/DataGrid/DataGridActions.svelte';
+  import SingleActionDataGrid from '../ui/DataGrid/SingleActionDataGrid.svelte';
+  import Panel from '../ui/Panel.svelte';
+  import SectionTitle from '../ui/SectionTitle.svelte';
+
+  export let dictionaries: DictionaryType[];
+  // Used when the table is in single select mode.
+  export let dictionaryId: number | null = null;
+  // Used when the table is in multiselect mode.
+  export let multiSelectDictionaryIds: Record<number, boolean> = {};
+  export let isEditingParcel: boolean = false;
+  export let isMultiselect: boolean = false;
+  export let hasEditPermission: boolean = false;
+  export let type: string;
+  export let user: User | null;
+
+  let dictionaryDataGrid: SingleActionDataGrid<DictionaryType> | undefined = undefined;
+  let dictionaryColumnDefs: DataGridColumnDef[];
+  let displayText: string = '';
+  let displayTextPlural: string = '';
+  let hasDeletePermission: boolean = false;
+  let isSequenceAdaptation: boolean = false;
+  let sequenceAdaptationColumDefs: DataGridColumnDef[];
+
+  const dispatch = createEventDispatcher<{
+    delete: { id: number };
+    multiSelect: { ids: Record<number, boolean> };
+    select: { id: number | null };
+  }>();
+
+  type CellRendererParams = {
+    deleteDictionary?: (dictionary: DictionaryType) => void;
+  };
+  type DictionaryCellRendererParams = ICellRendererParams<DictionaryType> & CellRendererParams;
+
+  $: isSequenceAdaptation = type === 'Sequence';
+  $: displayText = isSequenceAdaptation ? `${type} Adaptation` : `${type} Dictionary`;
+  $: displayTextPlural = isSequenceAdaptation ? `${type} Adaptations` : `${type} Dictionaries`;
+  $: hasDeletePermission = featurePermissions.commandDictionary.canDelete(user);
+
+  $: if ((multiSelectDictionaryIds || dictionaryId) && dictionaryDataGrid?.redrawRows !== undefined) {
+    dictionaryDataGrid.redrawRows();
+  }
+
+  const editingColumnDefs: DataGridColumnDef[] = [
+    {
+      cellDataType: 'boolean',
+      editable: hasEditPermission,
+      headerName: '',
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      valueGetter: (params: ValueGetterParams<DictionaryType>) => {
+        const { data } = params;
+        if (data) {
+          if (isMultiselect) {
```

Comment:
This would need to be in a reactive statement too to update what `isMultiselect` ever updates or the `dictionaryId`